### PR TITLE
feat: impl kcl-plugin framework with Go, and include the hello plugin

### LINKED ISSUE #48: feat: support Go write kcl plugin
## Enhancement

Go SDK should support develop kcl plugin with Go

### PR DESCRIPTION
**Pull request by chai2010**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y fix #48 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [X] Other

impl Go kcl-plugin framework, add include the hello plugin.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/kcl_plugin/hello_plugin/api.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	},
+}
+
+func argInt64(arg interface{}) int64 {
+	s := fmt.Sprint(arg)
+	v, _ := strconv.ParseInt(s, 10, 64)
+	return v
+}
+
+func parseArgs(args_json string) (args []interface{}, err error) {
```
Comment (Peefy) [inline pkg/kcl_plugin/hello_plugin/api.go]: `parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

Comment (Peefy) [reply pkg/kcl_plugin/hello_plugin/api.go]: > `parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

use KCL Value?

**Review by SparkYuan — DISMISSED** [review]

File: pkg/kcl_plugin/hello_plugin/api.go (hunk 1 above)
Comment (chai2010) [reply pkg/kcl_plugin/hello_plugin/api.go]: > `parseArgs` and `parseKwargs` should be called inside the kclvm-go-plugin framework and not exposed to the user?

remove the parseArgs and parseKwargs, add kcl_plugin.MethodArgs for args, and kcl_plugin.MethodResult for result

**Comment by chai2010** [comment]
PTAL

**Review by SparkYuan — DISMISSED** [review]
lgtm

File: pkg/kcl_plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	}
+
+	return _Invoke(method, args_json, kwargs_json)
+}
+
+func InvokeJson(method, args_json, kwargs_json string) (result_json string) {
+	return _Invoke(method, args_json, kwargs_json)
+}
+
+var limitChan_Invoke = make(chan struct{}, 4)
```
Comment (Peefy) [inline pkg/kcl_plugin/plugin.go]: Leave a todo on `4` or use a context variable? Hard writing 4 may not be the best option.

File: pkg/service/c_api_call/proto_capi_kclvm_service_client.go [hunk 3]
```diff
@@ -24,7 +25,7 @@ type PROTOCAPI_KclvmServiceClient struct {
 
 func PROTOCAPI_NewKclvmServiceClient() *PROTOCAPI_KclvmServiceClient {
 	c := new(PROTOCAPI_KclvmServiceClient)
-	c.client = NewKclvmService(C.longlong(0))
+	c.client = C.kclvm_service_new(C.longlong(kcl_plugin.GetInvokeJsonProxyPtr()))
```
Comment (Peefy) [inline pkg/service/c_api_call/proto_capi_kclvm_service_client.go]: s/C.kclvm_service_new/NewKclvmService

Comment (chai2010) [reply pkg/service/c_api_call/proto_capi_kclvm_service_client.go]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

File: pkg/kcl_plugin/plugin.go (hunk 2 above)
Comment (chai2010) [reply pkg/kcl_plugin/plugin.go]: remove the unused code

**Review by amyXia1994 — APPROVED** [review]
lgtm

**Pull request merged**